### ISSUE TITLE
Mac spire native binary hangs when run with ssh-agent active

### ISSUE THREAD

**Issue by retrogradeorbit** [epiccastle/spire#8]
Workaround: `export -n SSH_AUTH_SOCK` but then you can't use ssh-agent authentication

https://circleci.com/gh/epiccastle/spire/487

On JVM:

```
Agent pid 3171
Identity added: ssh_test_key_rsa (ssh_test_key_rsa)
Compiling 1 source files to /Users/distiller/project/target/default/classes
open-connection {:hostname "localhost", :port 2200, :key "localhost:2200", :host-string "localhost:2200"}
connect {:hostname "localhost", :port 2200, :key "localhost:2200", :host-string "localhost:2200"}
make-session #object[com.jcraft.jsch.JSch 0x2981a626 "com.jcraft.jsch.JSch@2981a626"] "localhost" {:hostname "localhost", :port 2200, :key "localhost:2200", :host-string "localhost:2200"}
make-session adding-option :hostname "localhost"
make-session adding-option :key "localhost:2200"
make-session adding-option :host-string "localhost:2200"
make-session returning #object[com.jcraft.jsch.Session 0x37a907cb "com.jcraft.jsch.Session@37a907cb"]
connect connecting
Connecting to localhost port 2200
Connection established
Remote version string: SSH-2.0-OpenSSH_7.4
Local version string: SSH-2.0-JSCH-0.1.54
CheckCiphers: aes256-ctr,aes192-ctr,aes128-ctr,aes256-cbc,aes192-cbc,aes128-cbc,3des-ctr,arcfour,arcfour128,arcfour256
aes256-ctr is not available.
aes192-ctr is not available.
aes256-cbc is not available.
aes192-cbc is not available.
CheckKexes: diffie-hellman-group14-sha1,ecdh-sha2-nistp256,ecdh-sha2-nistp384,ecdh-sha2-nistp521
CheckSignatures: ecdsa-sha2-nistp256,ecdsa-sha2-nistp384,ecdsa-sha2-nistp521
SSH_MSG_KEXINIT sent
SSH_MSG_KEXINIT received
kex: server: curve25519-sha256,curve25519-sha256@libssh.org,ecdh-sha2-nistp256,ecdh-sha2-nistp384,ecdh-sha2-nistp521,diffie-hellman-group-exchange-sha256,diffie-hellman-group16-sha512,diffie-hellman-group18-sha512,diffie-hellman-group14-sha256,diffie-hellman-group14-sha1
kex: server: ssh-rsa,rsa-sha2-512,rsa-sha2-256,ecdsa-sha2-nistp256,ssh-ed25519
kex: server: chacha20-poly1305@openssh.com,aes128-ctr,aes192-ctr,aes256-ctr,aes128-gcm@openssh.com,aes256-gcm@openssh.com
kex: server: chacha20-poly1305@openssh.com,aes128-ctr,aes192-ctr,aes256-ctr,aes128-gcm@openssh.com,aes256-gcm@openssh.com
kex: server: umac-64-etm@openssh.com,umac-128-etm@openssh.com,hmac-sha2-256-etm@openssh.com,hmac-sha2-512-etm@openssh.com,hmac-sha1-etm@openssh.com,umac-64@openssh.com,umac-128@openssh.com,hmac-sha2-256,hmac-sha2-512,hmac-sha1
kex: server: umac-64-etm@openssh.com,umac-128-etm@openssh.com,hmac-sha2-256-etm@openssh.com,hmac-sha2-512-etm@openssh.com,hmac-sha1-etm@openssh.com,umac-64@openssh.com,umac-128@openssh.com,hmac-sha2-256,hmac-sha2-512,hmac-sha1
kex: server: none,zlib@openssh.com
kex: server: none,zlib@openssh.com
kex: server: 
kex: server: 
kex: client: ecdh-sha2-nistp256,ecdh-sha2-nistp384,ecdh-sha2-nistp521,diffie-hellman-group14-sha1,diffie-hellman-group-exchange-sha256,diffie-hellman-group-exchange-sha1,diffie-hellman-group1-sha1
kex: client: ssh-rsa,ssh-dss,ecdsa-sha2-nistp256,ecdsa-sha2-nistp384,ecdsa-sha2-nistp521
kex: client: aes128-ctr,aes128-cbc,3des-ctr,3des-cbc,blowfish-cbc
kex: client: aes128-ctr,aes128-cbc,3des-ctr,3des-cbc,blowfish-cbc
kex: client: hmac-md5,hmac-sha1,hmac-sha2-256,hmac-sha1-96,hmac-md5-96
kex: client: hmac-md5,hmac-sha1,hmac-sha2-256,hmac-sha1-96,hmac-md5-96
kex: client: none
kex: client: none
kex: client: 
kex: client: 
kex: server->client aes128-ctr hmac-sha1 none
kex: client->server aes128-ctr hmac-sha1 none
SSH_MSG_KEX_ECDH_INIT sent
expecting SSH_MSG_KEX_ECDH_REPLY
ssh_rsa_verify: signature true
make-host-key-repository check "[localhost]:2200" #object["[B" 0x2f71cdc0 "[B@2f71cdc0"]
make-host-key-repository getHostKey "[localhost]:2200" "ssh-rsa"
make-host-key-repository getHostKey returning #object["[Lcom.jcraft.jsch.HostKey;" 0x41d9b9de "[Lcom.jcraft.jsch.HostKey;@41d9b9de"]
Host 'localhost' is known and matches the RSA host key
SSH_MSG_NEWKEYS sent
SSH_MSG_NEWKEYS received
SSH_MSG_SERVICE_REQUEST sent
SSH_MSG_SERVICE_ACCEPT received
Authentications that can continue: publickey,keyboard-interactive,password
Next authentication method: publickey
make-identity-repository getIdentities
open-auth-socket "/var/folders/ms/xg67k5sn16xc7sdr_w3q45840000gn/T//ssh-KZgCOtFfhVny/agent.3170"
open-auth-socket returning 62
request-identities 62
request-identities sending 11 [11]
send-query 62 (0 0 0 1 11) #object["[B" 0x134a27c1 "[B@134a27c1"] 5
request-identities bytes-written 5
request-identities read-response
request-identities response-read
request-identities reading 308 bytes
make-identity-repository close-auth-socket
close-auth-socket 62
make-identity-repository close-auth-socket closed
make-identity getPublicKeyBlob
make-identity getAlgName
make-identity isEncrypted
make-identity isEncrypted
make-identity setPassphrase
make-identity isEncrypted
make-identity getAlgName
make-identity getSignature #object["[B" 0x6015326 "[B@6015326"]
open-auth-socket "/var/folders/ms/xg67k5sn16xc7sdr_w3q45840000gn/T//ssh-KZgCOtFfhVny/agent.3170"
open-auth-socket returning 62
send-query 62 (0 0 2 156 13 0 0 1 23 0 0 0 7 115 115 104 45 114 115 97 0 0 0 3 1 0 1 0 0 1 1 0 -63 -64 -59 114 98 116 58 78 94 27 28 59 -85 -39 47 73 55 96 -60 77 -83 13 -46 -27 -54 -2 -54 -94 -109 38 56 60 -31 13 107 -9 64 60 45 127 100 -99 31 -43 -3 -42 -38 99 78 112 -46 6 1 -14 123 11 48 -49 -93 -96 -112 5 118 -123 18 -78 42 -70 118 -3 76 -124 102 58 -76 -50 24 83 46 120 -118 -104 96 -87 -10 -119 93 50 71 -51 22 -75 58 -111 26 -90 -118 -71 7 37 102 -119 -109 -32 109 87 29 127 120 -94 74 -34 -87 42 24 91 10 -20 80 47 -106 -33 -99 76 112 -30 113 9 89 -8 66 -126 118 21 -114 -30 -18 27 45 15 -59 60 -9 -88 -23 111 -107 25 9 58 63 74 109 121 -39 39 46 78 113 -116 2 -33 113 49 49 -12 42 59 74 -1 -57 88 69 93 15 -113 95 -34 -2 -122 83 -85 -80 -39 -109 20 69 61 -112 17 -52 89 -17 -123 90 97 126 115 -25 -110 105 56 -14 24 -27 13 92 -88 -103 110 -86 81 -98 21 115 69 24 122 62 -6 -4 -91 -68 19 -83 -52 -6 83 -51 26 46 -76 -80 -58 -95 24 4 37 114 -85 -37 75 -28 75 56 82 -27 41 -22 -123 9 80 120 59 122 -21 0 0 1 120 0 0 0 32 -11 46 116 106 -26 1 49 127 102 -4 -78 -54 84 -18 97 36 75 46 87 -109 43 16 18 -57 -121 -125 -68 -9 -111 79 -70 12 50 0 0 0 9 100 105 115 116 105 108 108 101 114 0 0 0 14 115 115 104 45 99 111 110 110 101 99 116 105 111 110 0 0 0 9 112 117 98 108 105 99 107 101 121 1 0 0 0 7 115 115 104 45 114 115 97 0 0 1 23 0 0 0 7 115 115 104 45 114 115 97 0 0 0 3 1 0 1 0 0 1 1 0 -63 -64 -59 114 98 116 58 78 94 27 28 59 -85 -39 47 73 55 96 -60 77 -83 13 -46 -27 -54 -2 -54 -94 -109 38 56 60 -31 13 107 -9 64 60 45 127 100 -99 31 -43 -3 -42 -38 99 78 112 -46 6 1 -14 123 11 48 -49 -93 -96 -112 5 118 -123 18 -78 42 -70 118 -3 76 -124 102 58 -76 -50 24 83 46 120 -118 -104 96 -87 -10 -119 93 50 71 -51 22 -75 58 -111 26 -90 -118 -71 7 37 102 -119 -109 -32 109 87 29 127 120 -94 74 -34 -87 42 24 91 10 -20 80 47 -106 -33 -99 76 112 -30 113 9 89 -8 66 -126 118 21 -114 -30 -18 27 45 15 -59 60 -9 -88 -23 111 -107 25 9 58 63 74 109 121 -39 39 46 78 113 -116 2 -33 113 49 49 -12 42 59 74 -1 -57 88 69 93 15 -113 95 -34 -2 -122 83 -85 -80 -39 -109 20 69 61 -112 17 -52 89 -17 -123 90 97 126 115 -25 -110 105 56 -14 24 -27 13 92 -88 -103 110 -86 81 -98 21 115 69 24 122 62 -6 -4 -91 -68 19 -83 -52 -6 83 -51 26 46 -76 -80 -58 -95 24 4 37 114 -85 -37 75 -28 75 56 82 -27 41 -22 -123 9 80 120 59 122 -21 0 0 0 0) #object["[B" 0x6d36529a "[B@6d36529a"] 672
close-auth-socket 62
Authentication succeeded (publickey).
connect connected
```

On Native-Image:

```
Agent pid 3349
Identity added: ssh_test_key_rsa (ssh_test_key_rsa)
open-connection {:hostname "localhost", :port 2200, :key "localhost:2200", :host-string "localhost:2200"}
connect {:hostname "localhost", :port 2200, :key "localhost:2200", :host-string "localhost:2200"}
make-session #object[com.jcraft.jsch.JSch 0x300ef446 "com.jcraft.jsch.JSch@10323c250"] "localhost" {:hostname "localhost", :port 2200, :key "localhost:2200", :host-string "localhost:2200"}
make-session adding-option :hostname "localhost"
make-session adding-option :key "localhost:2200"
make-session adding-option :host-string "localhost:2200"
make-session returning #object[com.jcraft.jsch.Session 0x54220d6f "com.jcraft.jsch.Session@10324fef8"]
connect connecting
Connecting to localhost port 2200
Connection established
Remote version string: SSH-2.0-OpenSSH_7.4
Local version string: SSH-2.0-JSCH-0.1.54
CheckCiphers: aes256-ctr,aes192-ctr,aes128-ctr,aes256-cbc,aes192-cbc,aes128-cbc,3des-ctr,arcfour,arcfour128,arcfour256
CheckKexes: diffie-hellman-group14-sha1,ecdh-sha2-nistp256,ecdh-sha2-nistp384,ecdh-sha2-nistp521
CheckSignatures: ecdsa-sha2-nistp256,ecdsa-sha2-nistp384,ecdsa-sha2-nistp521
ecdsa-sha2-nistp256 is not available.
ecdsa-sha2-nistp384 is not available.
ecdsa-sha2-nistp521 is not available.
SSH_MSG_KEXINIT sent
SSH_MSG_KEXINIT received
kex: server: curve25519-sha256,curve25519-sha256@libssh.org,ecdh-sha2-nistp256,ecdh-sha2-nistp384,ecdh-sha2-nistp521,diffie-hellman-group-exchange-sha256,diffie-hellman-group16-sha512,diffie-hellman-group18-sha512,diffie-hellman-group14-sha256,diffie-hellman-group14-sha1
kex: server: ssh-rsa,rsa-sha2-512,rsa-sha2-256,ecdsa-sha2-nistp256,ssh-ed25519
kex: server: chacha20-poly1305@openssh.com,aes128-ctr,aes192-ctr,aes256-ctr,aes128-gcm@openssh.com,aes256-gcm@openssh.com
kex: server: chacha20-poly1305@openssh.com,aes128-ctr,aes192-ctr,aes256-ctr,aes128-gcm@openssh.com,aes256-gcm@openssh.com
kex: server: umac-64-etm@openssh.com,umac-128-etm@openssh.com,hmac-sha2-256-etm@openssh.com,hmac-sha2-512-etm@openssh.com,hmac-sha1-etm@openssh.com,umac-64@openssh.com,umac-128@openssh.com,hmac-sha2-256,hmac-sha2-512,hmac-sha1
kex: server: umac-64-etm@openssh.com,umac-128-etm@openssh.com,hmac-sha2-256-etm@openssh.com,hmac-sha2-512-etm@openssh.com,hmac-sha1-etm@openssh.com,umac-64@openssh.com,umac-128@openssh.com,hmac-sha2-256,hmac-sha2-512,hmac-sha1
kex: server: none,zlib@openssh.com
kex: server: none,zlib@openssh.com
kex: server: 
kex: server: 
kex: client: ecdh-sha2-nistp256,ecdh-sha2-nistp384,ecdh-sha2-nistp521,diffie-hellman-group14-sha1,diffie-hellman-group-exchange-sha256,diffie-hellman-group-exchange-sha1,diffie-hellman-group1-sha1
kex: client: ssh-rsa,ssh-dss
kex: client: aes128-ctr,aes128-cbc,3des-ctr,3des-cbc,blowfish-cbc,aes192-ctr,aes192-cbc,aes256-ctr,aes256-cbc
kex: client: aes128-ctr,aes128-cbc,3des-ctr,3des-cbc,blowfish-cbc,aes192-ctr,aes192-cbc,aes256-ctr,aes256-cbc
kex: client: hmac-md5,hmac-sha1,hmac-sha2-256,hmac-sha1-96,hmac-md5-96
kex: client: hmac-md5,hmac-sha1,hmac-sha2-256,hmac-sha1-96,hmac-md5-96
kex: client: none
kex: client: none
kex: client: 
kex: client: 
kex: server->client aes128-ctr hmac-sha1 none
kex: client->server aes128-ctr hmac-sha1 none
SSH_MSG_KEX_ECDH_INIT sent
expecting SSH_MSG_KEX_ECDH_REPLY
ssh_rsa_verify: signature true
make-host-key-repository check "[localhost]:2200" #object["[B" 0x656f45cc "[B@1033bff18"]
make-host-key-repository getHostKey "[localhost]:2200" "ssh-rsa"
make-host-key-repository getHostKey returning #object["[Lcom.jcraft.jsch.HostKey;" 0x57216b97 "[Lcom.jcraft.jsch.HostKey;@1048262f8"]
Host 'localhost' is known and matches the RSA host key
SSH_MSG_NEWKEYS sent
SSH_MSG_NEWKEYS received
SSH_MSG_SERVICE_REQUEST sent
SSH_MSG_SERVICE_ACCEPT received
Authentications that can continue: publickey,keyboard-interactive,password
Next authentication method: publickey
make-identity-repository getIdentities
open-auth-socket "/var/folders/ms/xg67k5sn16xc7sdr_w3q45840000gn/T//ssh-0z9qOiXKxRum/agent.3348"
open-auth-socket returning 9
request-identities 9
request-identities sending 11 [11]
send-query 9 (0 0 0 1 11) #object["[B" 0x36c7da58 "[B@104b495b0"] 5
request-identities bytes-written 0
request-identities read-response
make: *** [macos-client-test] Terminated: 15

Too long with no output (exceeded 1m0s): context deadline exceeded
```

**Comment by retrogradeorbit** [comment]
https://github.com/epiccastle/graal-jni-unix-socket-test

**Comment by retrogradeorbit** [comment]
https://github.com/oracle/graal/issues/2152

**Comment by retrogradeorbit** [comment]
waiting now for https://github.com/oracle/graal/commit/4653525c0f1b135c51aef01c03080dfb961aba91 to be released in dev builds

**Comment by retrogradeorbit** [comment]
patch landed in https://github.com/graalvm/graalvm-ce-dev-builds/releases/tag/20.1.0-dev-20200225_0946
builds passing:
java8: https://circleci.com/gh/epiccastle/graal-jni-unix-socket-test/105
java11: https://circleci.com/gh/epiccastle/graal-jni-unix-socket-test/101

**Comment by retrogradeorbit** [comment]
fixed in 0.1.0-alpha.1 

https://circleci.com/gh/epiccastle/spire/591 : 2nd "Client Test" shows mac binary not hanging!
